generic template

SupinePandora43 opened the pull request:
for #1

SupinePandora43 committed:
generic template

SupinePandora43 committed:
fix template duplication

SupinePandora43 committed:
some naming

SupinePandora43 committed:
tab indentation, ignore .vs folder

SupinePandora43 committed:
TemplateModule.cs and TemplateModule.class

GlebChili commented on ProjectTemplates/gmodnet-module/TemplateModule.csproj:
```diff
@@ -0,0 +1,11 @@
+	<PropertyGroup>
+		<OutputType>Library</OutputType>
+		<TargetFramework>net5.0</TargetFramework>
+		<LangVersion>9.0</LangVersion>
```
We don't need to specify language version. It is 9.0 for .NET 5 by default.

SupinePandora43 replied:
resolved in https://github.com/GmodNET/Templates/pull/2/commits/b35fbb5c2afa30db85c220f50bde175673437d7c

GlebChili commented on ProjectTemplates/gmodnet-module/TemplateModule.csproj:
```diff
@@ -0,0 +1,11 @@
+	</PropertyGroup>
+
+	<ItemGroup>
+		<PackageReference Include="GmodNET.API" Version="0.7.0-beta.2.28828564.master" />
```
Fixing on some particular version of nighty build is suboptimal. Can we try wildcard `0.7.0-*`?

SupinePandora43 replied:
resolved in https://github.com/GmodNET/Templates/pull/2/commits/b35fbb5c2afa30db85c220f50bde175673437d7c

GlebChili commented on Template.sln:
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.30803.129
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Template", "Template.csproj", "{8A3CECC5-7801-4AB6-B549-F4AE8EFA1D59}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{8A3CECC5-7801-4AB6-B549-F4AE8EFA1D59}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{8A3CECC5-7801-4AB6-B549-F4AE8EFA1D59}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{8A3CECC5-7801-4AB6-B549-F4AE8EFA1D59}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{8A3CECC5-7801-4AB6-B549-F4AE8EFA1D59}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
+		SolutionGuid = {9A109073-21FC-46C6-A40A-8BE4AD21C24A}
+	EndGlobalSection
+EndGlobal
```
Why do we need .sln file at all? I think it should be removed.

SupinePandora43 replied:
resolved in https://github.com/GmodNET/Templates/pull/2/commits/b35fbb5c2afa30db85c220f50bde175673437d7c

SupinePandora43 committed:
remove solution file\ndo not specify language version\nGmodNET.API 0.7.0-* wildcard

GlebChili commented on Template.csproj:
```diff
@@ -0,0 +1,19 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<PropertyGroup>
+		<TargetFramework>net5</TargetFramework>
+		<PackageType>Template</PackageType>
+		<PackageVersion>1.0.0</PackageVersion>
+		<PackageId>GmodNET.Templates</PackageId>
+		<Title>GmodDotNet Templates</Title>
+		<Authors>GmodNET</Authors>
+		<Description>Project Templates for GmodDotNet module</Description>
+		<PackageTags>dotnet-new;templates</PackageTags>
+		<IncludeContentInPack>true</IncludeContentInPack>
+		<IncludeBuildOutput>false</IncludeBuildOutput>
+		<ContentTargetFolders>content</ContentTargetFolders>
+	</PropertyGroup>
```
Add `<NoBuild>true</NoBuild>` property. This will disable build at all.

GlebChili replied:
https://docs.microsoft.com/ru-ru/nuget/reference/msbuild-targets#packing-using-a-nuspec for more info.

SupinePandora43 replied:
resolved by https://github.com/GmodNET/Templates/pull/2/commits/5820dfeee45bead39c96e3adf9658f01ca7c7376

GlebChili commented on Template.csproj:
```diff
@@ -0,0 +1,19 @@
+	</PropertyGroup>
+	<ItemGroup>
+		<Content Include="ProjectTemplates\**\*" Exclude="ProjectTemplates\**\bin\**;ProjectTemplates\**\obj\**" />
+    	<Compile Remove="**\*" />
```
Do not need this if `<NoBuild>true</NoBuild>` is added.

SupinePandora43 replied:
resolved by https://github.com/GmodNET/Templates/pull/2/commits/5820dfeee45bead39c96e3adf9658f01ca7c7376

GlebChili commented on Template.csproj:
```diff
@@ -0,0 +1,19 @@
+		<ContentTargetFolders>content</ContentTargetFolders>
+	</PropertyGroup>
+	<ItemGroup>
+		<Content Include="ProjectTemplates\**\*" Exclude="ProjectTemplates\**\bin\**;ProjectTemplates\**\obj\**" />
```
Replace wildcards with explicit list of files. This will prevent unwanted objects from getting into the package.

SupinePandora43 replied:
@GlebChili, but it's for easier way adding new templates

GlebChili replied:
Still we need list them explicitly.

SupinePandora43 replied:
Resolved

SupinePandora43 committed:
use NoBuild

SupinePandora43 committed:
explicit files

GlebChili commented on ProjectTemplates/gmodnet-module/TemplateModule.csproj:
```diff
@@ -0,0 +1,10 @@
+		<TargetFramework>net5.0</TargetFramework>
+	</PropertyGroup>
+
+	<ItemGroup>
+		<PackageReference Include="GmodNET.API" Version="0.7.0-*" />
+	</ItemGroup>
```
Add `<FrameworkReference Include="Microsoft.AspNetCore.App" />` to ItemGroup. It will allow module crerators to use ASP.NET Core library in the module.

SupinePandora43 replied:
can it be added in asp net core module template instead?

GlebChili replied:
We will still need additional template, but I want user to be able ASP.NET Core types from any module.

SupinePandora43 replied:
resolved in https://github.com/GmodNET/Templates/pull/2/commits/bac331186d15a8074c5ebc1c9f4ea1d983b12efb

GlebChili commented on ProjectTemplates/gmodnet-module/.template.config/template.json:
```diff
@@ -0,0 +1,14 @@
+	"author": "SupinePandora43",
+	"classifications": [ "Common", "Library", "C#9" ],
+	"identity": "GmodNET.Template.module",
+	"name": "GmodDotNet Module Template",
```
Name should be GmodDotNet Module, not GmodDotNet Module Template.

SupinePandora43 replied:
resolved by https://github.com/GmodNET/Templates/pull/2/commits/43b3e391b40db967d94c5f16ea10354c3fa3c28f

GlebChili commented on ProjectTemplates/gmodnet-module/.template.config/template.json:
```diff
@@ -0,0 +1,14 @@
+	"$schema": "http://json.schemastore.org/template",
+	"author": "SupinePandora43",
+	"classifications": [ "Common", "Library", "C#9" ],
+	"identity": "GmodNET.Template.module",
```
Identity must be gmodnet-module.

SupinePandora43 replied:
resolved by https://github.com/GmodNET/Templates/pull/2/commits/43b3e391b40db967d94c5f16ea10354c3fa3c28f

GlebChili commented on ProjectTemplates/gmodnet-module/.template.config/template.json:
```diff
@@ -0,0 +1,14 @@
+﻿{
+	"$schema": "http://json.schemastore.org/template",
+	"author": "SupinePandora43",
+	"classifications": [ "Common", "Library", "C#9" ],
```
Classification must mention gmod-dot-net and gmodnet

SupinePandora43 replied:
resolved by https://github.com/GmodNET/Templates/pull/2/commits/43b3e391b40db967d94c5f16ea10354c3fa3c28f

GlebChili commented on ProjectTemplates/gmodnet-module/.template.config/template.json:
```diff
@@ -0,0 +1,14 @@
+﻿{
+	"$schema": "http://json.schemastore.org/template",
+	"author": "SupinePandora43",
```
Author must be GmodNET

SupinePandora43 replied:
resolved by https://github.com/GmodNET/Templates/pull/2/commits/43b3e391b40db967d94c5f16ea10354c3fa3c28f

SupinePandora43 committed:
change author from SupinePandora43 to GmodNET
change identity from GmodNET.Template.module to gmodnet-module
format (removed space)

SupinePandora43 committed:
add aspnetcore
